core/Fetch.cpp

```diff
@@ -49,52 +70,156 @@ namespace olympia
                                                          workload->getValueAsString(),
                                                          skip_nonuser_mode_);
 
-        fetch_inst_event_->schedule(1);
+        ev_fetch_insts->schedule(1);
     }
 
+
     void Fetch::fetchInstruction_()
     {
-        const uint32_t upper = std::min(credits_inst_queue_, num_insts_to_fetch_);
+        // Prefill the ibuf with some instructions read from the tracefile
+        // keeping enough capacity to group them into cache block accesses.
+        for (uint32_t i = ibuf_.size(); i < ibuf_capacity_; ++i)
+        {
+            const auto & inst_ptr = inst_generator_->getNextInst(my_clk_);
+            if (SPARTA_EXPECT_TRUE(nullptr != inst_ptr)) {
+                ibuf_.emplace_back(inst_ptr);
```

Comment:
Wonder if you can append to the `ibuf_` as blocks instead of individual instructions...